packages/shared/src/components/tabs/TabList.tsx

```diff
@@ -47,12 +60,40 @@ function TabList({
         });
       }
     }
-  }, [autoScrollActive, activeTabRect, offset]);
+  }, [autoScrollActive, dimensions]);
+
+  useLayoutEffect(() => {
+    // get the active tab's rect and offset so that we can position the indicator
+    const activeTabRect = currentActiveTab.current?.getBoundingClientRect();
+    const offset = activeTabRect ? currentActiveTab?.current.offsetLeft : 0;
+    const indicatorOffset = activeTabRect
+      ? activeTabRect.width / 2 + offset
+      : 0;
+
+    setDimensions((current) => {
+      if (
+        current.activeTabRect === activeTabRect &&
+        current.offset === offset &&
+        current.indicatorOffset === indicatorOffset
+      ) {
+        return current;
+      }
```

Comment:
Why do we need to do this within set state, when actually we're setting the same state? We should better do this check before

Reply:
I was contemplating on this, and here's my reasoning:
1. The `dimension` state itself is an object. So to actually set the state, we need to send it a new reference, which we are doing below
3. To avoid unnecessarily setting a reference with the same information, we need to check if the current state is different than what we want to set. We have 2 options:
  a. Check the state before calling the state setter and if it's the same, do not call it. For this we would need to add the `dimension` to the `useLayoutEffect` dependencies, which would mean that this function would be called on each update of state. Which seems unnecessary.
  b. Send a callback to state setter and make the check there.
  
I decided for option b, since it seems cleaner to me from the execution point of view.

I agree, the code looks a bit unweildy this way, but it works. We could theoretically extract some of the logic to an external function to make it look nicer, but since it's just a few lines, I decided not to.

Let me know your thoughts :)

Reply:
don't really have a strong opinion on it, so we should be good either way